src/pages/Notification/NotificationPage.jsx

```diff
@@ -49,17 +57,124 @@ const NotificationPage = () => {
     setModalShow(true);
   };
 
+  //펀딩알림 클릭시 해당 펀딩 페이지로 이동
   const fundingNotiClick = () => {
     window.location.href = '/'; // 나중에 이동할 주소 넣어야함
   };
 
-  const filteredNotifications = notifications.filter((notification) => {
+  const filteredNotifications = notificationList.filter(() => {
     if (tag === '전체') return true;
-    if (tag === '친구' && notification.type === '친구') return true;
-    if (tag === '펀딩' && notification.type === '펀딩') return true;
+    if (tag === '친구') return true;
+    if (tag === '펀딩') return true;
     return false;
   });
 
+  useEffect(() => {
+    if (tag === '전체') {
+      readNotificationAll();
+    } else if (tag === '친구') {
+      readNotificationFriend();
+    } else if (tag === '펀딩') {
+      readNotificationFunding();
+    }
+  }, [tag]);
+
+  //전체알림 조회 API 연결
+  const readNotificationAll = async () => {
+    try {
+      const response = await getAllNotice();
+      setNotificationList(formatNotifications(response.data));
+      console.log(response.data);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+  //친구알림 조회 API 연결
+  const readNotificationFriend = async () => {
+    try {
+      const response = await getFriendNotice();
+      const data = Array.isArray(response.data) ? response.data : [];
+      if (data.length === 0) {
+        setNotificationList([]);
+      } else
+        setNotificationList(
+          response.data.map((item) => ({
+            ...item,
+            time: getTime(item.updatedAt),
+            image: item.recieveUserImgUrl,
+            name: item.recieveUserNickname,
+            friendTableId: setFriendTableId(item.friendTableId),
+          })),
+        );

```

Comment:
알림 데이터 받아올 때 라벨링을 합니다.
그리고 state 기본값으로 이미 빈 배열 `[]`을 설정해두었으니 이렇게만 작성해도 됩니다.
```suggestion
      const friendNotifications = [
        response.data && response.data.map((item) => ({
            tag: '친구',
            friendTableId: setFriendTableId(item.friendTableId),
            name: item.recieveUserNickname,
            image: item.recieveUserImgUrl,
            time: getTime(item.updatedAt),
            ...item,
        }))
      ];
      setNotificationList(friendNotifications);
```